python/ray/autoscaler/subprocess_output_util.py

```diff
@@ -206,24 +224,31 @@ def _run_and_process_output(cmd,
     are read-only except for return values and possible exceptions.
     """
 
+    stdin_overwrite = subprocess.PIPE
+    assert not (does_allow_interactive() and is_output_redirected()), (
+        "Cannot redirect output while in interactive mode.")
+    if does_allow_interactive() and not is_output_redirected():
+        stdin_overwrite = None
+
+
+    # See implementation note #1
+
     if use_login_shells:
-        # See implementation note #1
         if stdout_file is None:
             stdout_file = subprocess.DEVNULL
         if stderr_file is None:
             stderr_file = subprocess.DEVNULL
```

Comment:
I reverted these changes; I think the behavior as-is is correct and the adjusted version (where if stderr_file is None, stderr_file=stdout_file) is incorrect.